readthedocs/search/documents.py

```diff
@@ -50,12 +50,14 @@ class ProjectDocument(RTDDocTypeMixin, Document):
 
     def get_queryset(self):
         """
+        Additional filtering of default queryset.
+
         Don't include delisted projects.
         This will also break in-doc search for these projects,
         but it's not a priority to find a solution for this as long as "delisted" projects are
         understood to be projects with a negative reason for being delisted.
         """
-        return super().get_queryset().exclude(delisted=True)
+        return super().get_queryset().exclude(delisted=True).exclude(is_spam=True)
```

Comment:
This is only our explicit spam setting, we need to filter by the rules:

https://github.com/readthedocs/readthedocs.org/blob/7c4f5c5cb5bc7eeaf46fc5ee12288be602a16dbf/readthedocs/projects/admin.py#219-L223

Reply:
it is on purpose, please see my comment in the issue for excluding spam projects from search: https://github.com/readthedocs/readthedocs-ext/issues/461#issuecomment-1444063694

Reply:
We should not use the `is_spam` field here, but a threshold as Eric mentioned.

Reply:
Let's discuss in a follow up 👍

Reply:
Judging from the discussion in https://github.com/readthedocs/readthedocs-ext/issues/461#issuecomment-1444063694, we can come up with a simple solution that probably adjusts this line further to exclude projects with a spam score.

But since that issue already tracks this, I'll leave it there. It's pretty easy to implement in a follow-up, AFAICT.